Framework/Algorithms/test/IntegrationTest.h

```diff
@@ -701,6 +701,107 @@ class IntegrationTest : public CxxTest::TestSuite {
     }
   }
 
+  void testStartWsIndexOutOfBounds() {
+    const int startIndex = 100;
+    const int endIndex = 4;
+
+    MatrixWorkspace_sptr input;
+    TS_ASSERT_THROWS_NOTHING(
+        input = AnalysisDataService::Instance().retrieveWS<MatrixWorkspace>(
+            "testSpace"));
+
+    Integration alg;
+    alg.setRethrows(true);
+    TS_ASSERT_THROWS_NOTHING(alg.initialize());
+    TS_ASSERT_THROWS_NOTHING(
+        alg.setPropertyValue("InputWorkspace", "testSpace"));
+    TS_ASSERT_THROWS_NOTHING(alg.setPropertyValue("OutputWorkspace", "out"));
+    TS_ASSERT_THROWS_NOTHING(
+        alg.setProperty("StartWorkspaceIndex", startIndex));
+    TS_ASSERT_THROWS_NOTHING(alg.setProperty("EndWorkspaceIndex", endIndex));
```

Comment:
...to here could be moved to a function which accepts an input workspace name, output workspace name, start index and end index. This avoids a lot of the code repetition